azurerm_oracle_resource_anchor 'name' validation missing hyphen '-'

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.15.5

### AzureRM Provider Version

4.74.0

### Affected Resource(s)/Data Source(s)

azurerm_oracle_resource_anchor

### Terraform Configuration Files

```hcl
resource "azurerm_oracle_resource_anchor" "anchor" {
  name                = "Bla-Bla-Bla-Bla-Bla-01"
  resource_group_name = "blablaRsg"
}
```

### Debug Output/Panic Output

```shell
│ Error: name must contain only letters and numbers
│ 
│   with azurerm_oracle_resource_anchor.ra-odaa,
│   on main.tf line 82, in resource "azurerm_oracle_resource_anchor" "ra-odaa":
│   82:   name                = "Bla-Bla-Bla-Bla-Bla-01"
│ 
╵
Error: Process completed with exit code 1.
```

### Expected Behaviour

It must allow the "hyphen" in the name as the azure api allows it.

<img width="717" height="69" alt="Image" src="https://github.com/user-attachments/assets/4d498978-e9d5-4da5-b449-00f463eaa48a" />

<img width="519" height="107" alt="Image" src="https://github.com/user-attachments/assets/a17777d3-e2d7-427e-adb8-87785596f2e2" />


### Actual Behaviour


│ Error: name must contain only letters and numbers
│ 
│   with azurerm_oracle_resource_anchor.ra-odaa,
│   on main.tf line 82, in resource "azurerm_oracle_resource_anchor" "ra-odaa":
│   82:   name                = "Bla-Bla-Bla-Bla-Bla-01"
│ 
╵
Error: Process completed with exit code 1.

### Steps to Reproduce

terraform plan

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/oracle/validate/oracle_resource_anchor.go
+++ b/internal/services/oracle/validate/oracle_resource_anchor.go
@@ -19,13 +19,13 @@ func ResourceAnchorName(i interface{}, k string) (warnings []string, errors []er
 
 	hasInvalidChar := false
 	for _, r := range v {
-		if !unicode.IsLetter(r) && !unicode.IsNumber(r) {
+		if !unicode.IsLetter(r) && !unicode.IsNumber(r) && r != '-' {
 			hasInvalidChar = true
 			break
 		}
 	}
 	if hasInvalidChar {
-		validationErrors = append(validationErrors, fmt.Errorf("%v must contain only letters and numbers", k))
+		validationErrors = append(validationErrors, fmt.Errorf("%v must contain only letters , numbers and hyphens", k))
 	}
 
 	if len(v) > 24 {

